src/routes/accountOpeningRequest.ts

```diff
@@ -45,22 +55,28 @@ export const createAccountOpeningRequest = async (
     },
   };
 
-  const personKey = `reslock:${process.env.MOCKSOLARIS_REDIS_PREFIX}:person:${personId}`;
-  let person;
-  await redlock.using([personKey], 5000, async (signal) => {
+  const entityKey = `reslock:${
+    process.env.MOCKSOLARIS_REDIS_PREFIX
+  }:${data.customer_type.toLowerCase()}:${entityId}`;
+  let entity: MockPerson | MockBusiness;
+  await redlock.using([entityKey], 5000, async (signal) => {
     if (signal.aborted) {
       throw signal.error;
     }
-    person = await getPerson(personId);
-    person.accountOpeningRequests = person.accountOpeningRequests || [];
-    person.accountOpeningRequests.push(accountOpeningRequest);
-    await savePerson(person);
+    entity = await getEntity(entityId);
+    entity.accountOpeningRequests = entity.accountOpeningRequests || [];
+    entity.accountOpeningRequests.push(accountOpeningRequest);
+    await saveEntity(entity);
   });
-  await saveAccountOpeningRequestToPersonId(accountOpeningRequest.id, personId);
+  await saveAccountOpeningRequestToEntityId(
+    accountOpeningRequest.id,
+    entityId,
+    data.customer_type.toLowerCase()
+  );
 
   res.status(HttpStatusCodes.CREATED).send(accountOpeningRequest);
 
-  const account = await createAccount(personId);
+  const account = await createAccount(entityId, undefined, data.customer_type);
```

Comment:
you should pass proper AccountType in data instead of undefined, otherwise you will create default account type CHECKING_SOLE_PROPRIETOR for Business. But it should be CHECKING_BUSINESS

Reply:
how could i miss that